io-engine/src/bdev/nexus/nexus_bdev_rebuild.rs

```diff
@@ -126,7 +126,10 @@ impl<'n> Nexus<'n> {
         // rebuilt would then need to be rebuilt again.
         // Ensuring that the dst child receives all frontend Write IO keeps all
         // rebuilt ranges in sync with the other children.
-        self.reconfigure(DrEvent::ChildRebuild).await;
+        self.reconfigure(DrEvent::ChildRebuildBegin(
+            self.get_child_device_name(dst_child_uri.as_str())?,
```

Comment:
you mean the String containing device name? hmm, It should be bound to lifetime of DrEvent I guess, so when the TraverseCtx is destroyed, the contained DrEvent(with String) should be destroyed too. Beyond `nexus::reconfigure`, there is no error path present today.

Reply:
btw, today it's possible that if a rebuild fails and unplug is triggered, it'll reconfigure that channels and then later the `on_rebuild_update() `will again do the same which is redundant I believe. Now, as a result of rebuild failure, the unplug will reconfigure the channels and the `on_rebuild_update() `will bail out early because `get_child_device_name() `will fail  with `ChildNotOpen` error:

> failed to process rebuild update notification with error: Child nvmf://65.21.0.104:8420/nqn.2019-05.io.openebs:233532e2-4fe0-436f-9eaa-c4652bc53f58?uuid=233532e2-4fe0-436f-9eaa-c4652bc53f58 of nexus f8079f24-6fe5-4532-ac88-fa0cbc31cbb4 is not open

Reply:
No, I mean does the rebuild job get cleaned up by upper layers if we fail here?

Reply:
Looks like we need to call `child.remove_rebuild_job() `if at all this fails here. Will that drop the `RebuildJobBackEnd` as well or shall we explicitly do `self.comms.send(None)` before calling `child.remove_rebuild_job()`